client/.eslintrc.js

```diff
@@ -26,6 +26,17 @@ module.exports = {
 			},
 		],
 		'jest/no-mocks-import': 'off',
+		'no-restricted-syntax': [
+			'error',
+			{
+				selector: [
+					'CallExpression[callee.name="useSelector"][arguments.length=1] > ArrowFunctionExpression > :matches(ObjectExpression, ArrayExpression)',
+					'CallExpression[callee.name="useSelector"][arguments.length=1] > ArrowFunctionExpression > BlockStatement > ReturnStatement > :matches(ObjectExpression, ArrayExpression)',
```

Comment:
I discovered / remembered that `useSelector` accepts a second argument `equalityFn` to customize the default equality behavior, useful in cases where it's intentional to return an object that's evaluated by e.g. `shallowEqual`. 

Updated in 7ebfd6b1cbd to both allow this usage, and to include it as an alternative in the message.